Feat/glif 39 refactor multisig navigation

Schwartz10 opened the pull request.

erchaves committed:
Adding the react-components library.

Updating the readme to explain how `npm link` is used for local
development.

erchaves committed:
first pass -- adding a couple navigation buttons which go to url routes

erchaves committed:
first demonstration of changing views.

Note that this is currently using both urls and view changes in a hacky
way. We should probably decide which pattern to use and switch to just
that pattern.

erchaves committed:
clean up comments, add isActive state style

erchaves committed:
use new NavLink instead of NavButton

and use router instead of window.location — even though this is just
temp logic and will go away

erchaves committed:
adding utils/urlParams from a future branch.

Should be a trivial merge later

erchaves committed:
adding an owners page, updating logic to use page url constants

erchaves committed:
adding case for owners page to show accounts only on that owners page

erchaves committed:
lint fixes

erchaves committed:
fixing missing comma

erchaves committed:
First pass at refactoring MSIG and restyling it.

erchaves committed:
lint fixes

erchaves committed:
rename the lint fix helper that I added before. lint:fix better matches others

erchaves committed:
adding other pages

erchaves committed:
lint fix

erchaves committed:
lint updates

erchaves committed:
Merge branch 'primary' into feat/GLIF-39-refactor-multisig-navigation

erchaves committed:
adding close buttons to individual addresses

erchaves committed:
lint fixes

erchaves committed:
differentiating between onClose and onComplete calls

erchaves committed:
Merge branch 'primary' into feat/GLIF-39-refactor-multisig-navigation

# Conflicts:
#	components/Msig/Home/index.jsx

erchaves committed:
fix merge for msig/home/index

Schwartz10 committed:
Fix issue with input

Schwartz10 commented on components/Msig/AddRmSigners/Base.jsx:
```diff
@@ -356,13 +361,15 @@ const ManipulateSignersHOC = method => {
       })
-    )
+    ),
+    cid: ADDRESS_PROPTYPE
```
This should just be `PropTypes.string.isRequired`. A `cid` is not a filecoin address.

Schwartz10 commented on components/Msig/AddRmSigners/Base.jsx:
```diff
@@ -47,6 +47,11 @@ const ManipulateSignersHOC = method => {
+    // console.log(
+    //   `TODO: change UI auto select the id to remove. Selected: ${cid}`
+    // )
+
```
Let's make sure to get rid of this when you're ready

Schwartz10 commented on components/Msig/Home/AccountSummary.jsx:
```diff
@@ -15,137 +22,143 @@ import {
+            minWidth={9}
+            borderRadius={6}
+          />
+        )}
```
This doesn't go here anymore - the "edit" button alongside each Signer row is the new way to "change signer".

erchaves committed:
auto selecting the user to remove

Schwartz10 commented on components/Msig/Home/Address.jsx:
```diff
@@ -44,20 +52,19 @@ const Address = ({ address, label, glyphAcronym }) => {
-        />
+        <CopyAddress color='core.nearblack' address={address} />
       </Box>
+      {onClose && <ButtonClose role='button' type='button' onClick={onClose} />}
```
I understand why you're doing this, and I don't have any specific problem with approaching it this way, but it just _feels_ out of place to me for some reason. Not sure what to do yet, but can leave for now unless you think of something better.

erchaves replied:
oh, well, `onClose` is definitely the wrong name I'm seeing. We could rename this to removeAddress or something. But do you mean that the UI is also wrong? We can remove this if you want.

Schwartz10 replied:
Yeah i like `onRemoveAddress` (let's stick with `on__` if we're going that way). I haven't seen the UI let, need to add a signer to my multisig

Schwartz10 commented on components/Msig/Home/Balances.jsx:
```diff
@@ -1,10 +1,14 @@
 
+import { PAGE_MSIG_WITHDRAW } from '../../../constants'
+
+import { gotoRouteWithKeyUrlParams } from '../../../utils/urlParams'
```
Pedantic but lets just squeeze these together with no enters

Schwartz10 commented on components/Msig/Home/State.jsx:
```diff
@@ -1,19 +1,21 @@
-  MenuItem,
-  IconGlif
-} from '../../Shared'
+import { Box, Title, Menu, MenuItem, IconGlif } from '../../Shared'
```
Let's import things from `@glif/react-components` as much as we can!

erchaves replied:
nice, cool, ill switch all those

Schwartz10 replied:
i would make sure to do it slowly just so you know if something broke along the way (i clearly didn't follow this advice when importing inputs into the app)

Schwartz10 commented on components/Msig/Home/index.jsx:
```diff
@@ -33,50 +44,67 @@ const MsigHome = () => {
+            <ChangeOwner
+              close={() => {
+                gotoRouteWithKeyUrlParams(router, PAGE_MSIG_CHANGE_OWNER)
+              }}
```
Marking this as a place we'll need to stub the same onComplete, onClose functionality..

Schwartz10 commented on components/Msig/Home/index.jsx:
```diff
@@ -1,27 +1,38 @@
 import Withdraw from '../Withdraw'
 import ChangeOwner from '../ChangeOwner'
 import { AddSigner, RemoveSigner } from '../AddRmSigners'
 import { Box, LoadingScreen } from '../../Shared'
```
Let's try and import from the glif react components

Schwartz10 reviewed (commented):
A couple high level things:

1. I think we should do for each of the multisig pages what we did for the send page - remember how you ripped the send page out of the home page, and just wrapped it in the `PageWrapper` component you pulled into the Glif react components? Let's do that for each of these multisig pages, so we don't need to use MsigHome to render everything.
2. This should make the active/inactive navigation tabs quite easy, here's what I'd do:
- Create a new component inside `/components/Msig` called `Nav`
- `Nav` should render the `NavLinks`.
- `Nav` is responsible for reading the router via `useRouter` and passing the active prop to the `NavLink`
- `Nav` gets rendered on each page, separately
3. Please uncommit any files that don't have to do with with this PR - like small linter changes in files that you wouldn't normally have touched. This is a big PR as it is, so we should keep it to the necessary changes. A major linting fix is coming this weekend.
4. In any file you touch, let's _try_ to import as many components as we can from the @glif/react-component library. A good strategy for doing this is:
- Run the tests for that file (they should pass)
- Change the import from 'shared' to 'react-components'
- Run the tests for that file
- If the tests pass, great. integration complete
- If the tests fail, lets see which component it fails on, and create a new Jira ticket to assess the diff of the shared directory in the wallet vs the react component library (we already know from earlier today that the Input component needs an assessment.
5. The Owners page can now include the `Confirmation Threshold` part of the designs - you can access this through `const { NumApprovalsThreshold } = useMsig()`

erchaves committed:
adding onComplete in addition to onClose to other pages

Swapping out childView for detectPage. Consider using next.js’s files
to set the page instead. But this works fine.

erchaves committed:
Merge branch 'primary' into feat/GLIF-39-refactor-multisig-navigation

erchaves committed:
Fixing merge after react-components update. And Factoring NavMenu out into it's own component

erchaves committed:
Linting. And fixing name of MyNavLink

erchaves committed:
fix issue where this was opening in a new window which was breaking the functionality

It’s weird that this just started breaking because looking at all the
relevant files, none of them have been changed recently. So I'm not
sure when or how this breaking change happened. This probably isn't the
correct fix. So we probably want to figure out how to not have
target=‘_blank’ as a default.

erchaves committed:
cleanup and replacing shared components with glif/react-components

erchaves committed:
More switching from shared to glif/react-components

erchaves committed:
adding msig.NumApprovalsThreshold

erchaves committed:
lint cleanup. Change tooltip msg

Schwartz10 committed:
Cleanup/glif 39 (#854)

* Bump actions/setup-node from 2.1.5 to 2.4.0 (#849)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 2.1.5 to 2.4.0.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v2.1.5...v2.4.0)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump babel-eslint from 10.0.3 to 10.1.0 (#837)

Bumps [babel-eslint](https://github.com/babel/babel-eslint) from 10.0.3 to 10.1.0.
- [Release notes](https://github.com/babel/babel-eslint/releases)
- [Commits](https://github.com/babel/babel-eslint/compare/v10.0.3...v10.1.0)

---
updated-dependencies:
- dependency-name: babel-eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump object-path from 0.11.5 to 0.11.7 (#840)

Bumps [object-path](https://github.com/mariocasciaro/object-path) from 0.11.5 to 0.11.7.
- [Release notes](https://github.com/mariocasciaro/object-path/releases)
- [Commits](https://github.com/mariocasciaro/object-path/commits)

---
updated-dependencies:
- dependency-name: object-path
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump @testing-library/react from 10.0.1 to 12.0.0 (#836)

* Bump @testing-library/react from 10.0.1 to 12.0.0

Bumps [@testing-library/react](https://github.com/testing-library/react-testing-library) from 10.0.1 to 12.0.0.
- [Release notes](https://github.com/testing-library/react-testing-library/releases)
- [Changelog](https://github.com/testing-library/react-testing-library/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testing-library/react-testing-library/compare/v10.0.1...v12.0.0)

---
updated-dependencies:
- dependency-name: "@testing-library/react"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Fix tests

* Remove log

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jon Schwartz <jpschwartz2@uwalumni.com>

* Bump axios from 0.21.1 to 0.21.4 (#847)

Bumps [axios](https://github.com/axios/axios) from 0.21.1 to 0.21.4.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/master/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v0.21.1...v0.21.4)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Switch to next lint (#853)

* Set up next lint

* Update snapshots

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Bret <166301+bcomnes@users.noreply.github.com>

Schwartz10 commented:
Closing in favor of #855
